storage-provider/server/src/storage.rs

```
@@ -284,10 +293,7 @@ async fn upload(
     })?;
 
     if proposed_deal.piece_cid != piece_commitment_cid {
-        if let Err(err) = tokio::fs::remove_file(&file_path).await {
-            tracing::error!(%err, path = %file_path.display(), "failed to remove uploaded piece");
-        }
-
+        let _ = tokio::fs::remove_file(&file_path).await;
```

You're removing the log and not returning the error.
Silently failing and leaving the user in the dark *IS NOT OK*